src/Input/Extensions/Silk.NET.Input.Extensions/InputSnapshot.cs

```diff
@@ -21,25 +22,26 @@ internal InputSnapshot(IInputContext ctx)
             Mice = new MouseState[mice.Count];
             for (var i = 0; i < gamepads.Count; i++)
             {
-                Gamepads[i] = new GamepadState(gamepads[i]);
+                Gamepads[i] = new(gamepads[i], pool);
             }
 
             for (var i = 0; i < joysticks.Count; i++)
             {
-                Joysticks[i] = new JoystickState(joysticks[i]);
+                Joysticks[i] = new(joysticks[i], pool);
             }
 
             for (var i = 0; i < keyboards.Count; i++)
             {
-                Keyboards[i] = new KeyboardState(keyboards[i]);
+                Keyboards[i] = new(keyboards[i], pool);
             }
 
             for (var i = 0; i < mice.Count; i++)
             {
-                Mice[i] = new MouseState(mice[i]);
+                Mice[i] = new(mice[i], pool);
             }
         }
 
+        // TODO in 3.0 rent these too! right now the capture extensions are very expensive
         public GamepadState[] Gamepads { get; }
```

Comment:
another reason all public APIs should just use (RO)Memory

Reply:
Yep, that's exactly why this needs to be done in 3.0 because I really want to switch this over to use ROMemory